src/components/gift_list_page/GiftList/index.tsx

```diff
@@ -51,36 +51,19 @@ const ErrorText = styled.div`
   font-weight: 500;
 `;
 
-type Params = {
-  targetType: string;
-  rankType: string;
-};
-
 export const GiftList = () => {
-  const [giftItems, setGiftItems] = useState<GiftItemData[]>([]);
   const [isViewMore, setIsViewMore] = useState(false);
   const [targetType, setTargetType] = useState(localStorage.getItem('currentTarget') || 'ALL');
   const [rankType, setRankType] = useState(localStorage.getItem('currentTopic') || 'MANY_WISH');
-  const { data, isLoading, isError } = useQuery<
-    GiftItemData[],
-    Error,
-    GiftItemData[],
-    [string, Params]
-  >({
+  const { data, isLoading, isError } = useQuery({
     queryKey: ['giftItems', { targetType, rankType }],
-    queryFn: getGiftItems,
+    queryFn: ({ queryKey }: { queryKey: QueryKey }) => {
+      const { targetType, rankType } = queryKey[1];
+      if (!targetType || !rankType) throw new Error('arguments are required');
+      return getGiftItems(targetType, rankType);
+    },
```

Comment:
[참고] 만약 `queryKey`의 타입추론을 정확한 받고 싶다면 `queryKey` 배열을 `readonly`로 전달하시면 되요.

```suggestion
    queryKey: ['giftItems', { targetType, rankType}] as const,
    queryFn: ({ queryKey: [_, params] }) => {
      const { targetType, rankType } = params;
      /**
       * `queryKey`가 아래처럼 타입 추론됨
       * ```ts
       * interface params: {
       *   readonly targetType: string;
       *   readonly rankType: string;
       * }
       * ```
       */

      // 타입 가드 불필요
      // if (!targetType || !rankType) throw new Error('arguments are required');

      return getGiftItems(targetType, rankType);
    },
```